src/MacSrc/Shock.c

```diff
@@ -329,6 +329,9 @@ void SetSDLPalette(int index, int count, uchar *pal)
 
 void SDLDraw()
 {
+    extern void ZoomDrawProc(int erase);
+    ZoomDrawProc(FALSE); //draw zoom rectangle if enabled; if not, returns immediately
```

Comment:
These calls seem more fitting in one of the game loops than in the overall `SDLDraw` function

Reply:
I'll put them in mainloop.c.